auth/client/src/test/java/org/wildfly/security/auth/client/EncryptedExpressionXMLParserTest.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.client;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wildfly.security.SecurityFactory;
+import org.wildfly.security.credential.SecretKeyCredential;
+import org.wildfly.security.credential.store.CredentialStore;
+import org.wildfly.security.encryption.SecretKeyUtil;
+
+import javax.crypto.SecretKey;
+import java.io.File;
+import java.net.URL;
+import java.security.Provider;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.ServiceLoader;
+import java.util.function.Supplier;
+
+/**
+ * A test class to test the XML parser for Encrypted Expression Client Schema.
+ * @author <a href="mailto:prpaul@redhat.com">Prarthona Paul</a>
+ */
+
+public class EncryptedExpressionXMLParserTest {
+
+    private static final File CREDSTORE_DIR = new File("./target/credstore");
+    private static final String CLIENT_CREDSTORE_FILENAME = "/mycredstore.cs";
+
+    @Test
+    public void testEncryptedExpressionClient() throws Exception {
+        URL config = getClass().getResource("test-encrypted-expression-v1_0.xml");
+        System.setProperty("wildfly.config.url", config.getPath());
+
+        SecurityFactory<EncryptedExpressionContext> clientConfiguration = EncryptedExpressionsXmlParser.parseEncryptedExpressionClientConfiguration(config.toURI());
+        Assert.assertNotNull(clientConfiguration);
+        System.clearProperty("wildfly.config.url");
+    }
+
+    @Test
+    public void testEncryptedExpressionWithAuthClient() throws Exception {
+        URL config = getClass().getResource("test-encrypted-expression-auth-client-v1_0.xml");
+        System.setProperty("wildfly.config.url", config.getPath());
+
+        SecurityFactory<EncryptedExpressionContext> clientConfiguration = EncryptedExpressionsXmlParser.parseEncryptedExpressionClientConfiguration(config.toURI());
+        EncryptedExpressionConfig encExpConfig = clientConfiguration.create().encryptedExpressionConfig;
+        String encryptedExpression = encExpConfig.encryptedExpressionResolver.createExpression("password", encExpConfig);
+
+        //expression is encrypted during runtime, so it cannot be statically defined in client config file
+        System.setProperty("ENC_EXP_PROP", encryptedExpression);
+
+        SecurityFactory<AuthenticationContext> authClientConfiguration = ElytronXmlParser.parseAuthenticationClientConfiguration(config.toURI());
+        Assert.assertNotNull(clientConfiguration);
+        Assert.assertNotNull(authClientConfiguration);
```

Comment:
can we add some test that verifies that the resolved expression is as expected?

Reply:
Looking into this now